unstructured/partition/xlsx.py

```diff
@@ -243,6 +244,91 @@ def partition_xlsx(
     return elements
 
 
+class _SubtableParser:
+    """Distinguishes core-table from leading and trailing title rows in a subtable.
+
+    A *subtable* is a contiguous block of populated cells in the spreadsheet. Leading or trailing
+    rows of that block containing only one populated cell are called "single-cell rows" and are
+    not considered part of the core table. These are each emitted separately as a `Text`-subtype
+    element.
+    """
+
+    def __init__(self, subtable: pd.DataFrame):
+        self._subtable = subtable
+
+    @lazyproperty
+    def core_table(self) -> pd.DataFrame | None:
+        """The part between the leading and trailing single-cell rows, if any."""
+        core_table_start = len(self._leading_single_cell_row_indices)
+
+        # -- if core-table start is the end of table, there is no core-table
+        # -- (all rows are single-cell)
+        if core_table_start == len(self._subtable):
+            return None
+
+        # -- assert: there is at least one core-table row (leading single-cell rows greedily
+        # -- consumes all consecutive single-cell rows.
+
+        core_table_stop = len(self._subtable) - len(self._trailing_single_cell_row_indices)
+
+        # -- core-table is what's left in-between --
+        return self._subtable[core_table_start:core_table_stop]
+
+    def iter_leading_single_cell_rows_texts(self) -> Iterator[str]:
+        """Generate the cell-text for each leading single-cell row."""
+        for row_idx in self._leading_single_cell_row_indices:
+            yield self._subtable.iloc[row_idx].dropna().iloc[0]  # pyright: ignore
```

Comment:
`pandas` typing is poor so some of these calls generate multiple typing errors and must be ignored.